src/dbnode/persist/fs/seek_manager.go

```diff
@@ -500,25 +503,26 @@ func (m *seekerManager) UpdateOpenLease(
 	return updateLeaseResult, nil
 }
 
-func (m *seekerManager) startUpdateOpenLease(descriptor block.LeaseDescriptor) (bool, error) {
+func (m *seekerManager) startUpdateOpenLease(
+	namespace ident.ID,
+	hashableDescriptor block.HashableLeaseDescriptor,
+) (bool, error) {
 	m.Lock()
 	defer m.Unlock()
```

Comment:
nit: would we want to release this lock and perhaps use a `sync.Map` like we do in `block/lease.go` for tracking open leases? Although tbh perf wise this prob wouldn't make a diff at all.

Reply:
Using `sync.Map` (as I did in `block/lease.go`) was the initial approach that I tried, but then I realized that the lock is being held for the whole duration of `startUpdateOpenLease`, and there is this `namespace` check in between progress check / update, so I decided to go with a simple `map` under this lock (to better preserve the original semantics).